sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparison.scala

```diff
@@ -293,6 +299,34 @@ object UnwrapCastInBinaryComparison extends Rule[LogicalPlan] {
     }
   }
 
+  /**
```

Comment:
can we also update the doc for `UnwrapCastInBinaryComparison`? 

> * Currently this only handles cases where:
>  *   1). `fromType` (of `fromExp`) and `toType` are of numeric types (i.e., short, int, float, decimal, etc) or boolean type
>  *   2). `fromType` can be safely coerced to `toType` without precision loss (e.g., short to int, int to long, but not long to int, nor int to boolean)

This is no longer true.